### PR TITLE
python-rados: extract type-checking into a decorator

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <tchaikov@gmail.com>

### CODE REVIEW

File: src/test/pybind/test_rados.py [hunk 1]
```diff
@@ -38,6 +38,37 @@ def test_ioctx_context_manager():
...
+    def _method_plain(self, foo, bar, baz):
+        ok(isinstance(foo, str))
+        ok(isinstance(bar, int))
+        ok(isinstance(baz, int))
+        return (foo, bar, baz)
+
+    def test_method_plain(self):
+        assert_raises(TypeError, self._method_plain, 42, 42, 42)
+        assert_raises(TypeError, self._method_plain, '42', '42', '42')
+        assert_raises(TypeError, self._method_plain, '42', '42', '42')
```
Comment (jdurgin) [inline src/test/pybind/test_rados.py]: duplicate of previous line

**Comment by jdurgin** [comment]
Looks great! just noted a couple minor things

**Comment by dachary** [comment]
@tchaikov FYI, it needs a rebase ;-)

**Comment by tchaikov** [comment]
thanks @jdurgin for you review. sorry for the latency.

@dachary, i rebased against the latest master. thank you!

**Comment by dachary** [comment]
:+1:

**Comment by jdurgin** [comment]
LGTM

**Pull request merged**